zvariant_utils/src/case.rs

```diff
@@ -26,16 +26,60 @@ pub fn pascal_or_camel_case(s: &str, is_pascal_case: bool) -> String {
     result
 }
 
-/// Convert to snake case, assuming pascal case.
+/// Convert to snake case.
 ///
 /// If `s` is already in snake case, should yield the same result.
 pub fn snake_case(s: &str) -> String {
     let mut snake = String::new();
-    for ch in s.chars() {
+    for ch in pascal_or_camel_case(s, true).chars() {
```

Comment:
I'm not sure I'm able to follow the logic here. `pascal_or_camel_case` takes snake_case string so why would we call it on the input to this function, that's supposed to create snake case?

I think not doing multiple conversions will help with readability here.

Reply:
Calling `pascal_or_camel_case` just ensures that it processes PascalCase instead of assuming Pascal case, this is also why I removed the comment.

Reply:
Thanks for explaining! Yeah, now I think I get it. :laughing: However, I really think it would be easier to follow for a dumb person like myself to follow the code if each conversion did just one conversion directly.

Reply:
I re worked the functions and made it work similar to the `pascal_or_camal_case`. And don't have them depend on each other.